service/history/replicatorQueueProcessor.go

```diff
@@ -490,3 +546,29 @@ func convertLastReplicationInfo(info map[string]*persistence.ReplicationInfo) ma
 
 	return replicationInfoMap
 }
+
+func (p *replicatorQueueProcessorImpl) getTasks(readLevel int64) (*replicator.ReplicationTasksInfo, error) {
+	taskInfoList, hasMore, err := p.readTasks(readLevel)
+	if err != nil {
+		return nil, err
+	}
+
+	var replicationTasks []*replicator.ReplicationTask
+	for _, taskInfo := range taskInfoList {
+		readLevel = taskInfo.GetTaskID()
+		replicationTask, err := p.toReplicationTask(taskInfo)
+		if err != nil {
+			return nil, err
+		}
+
+		if replicationTask != nil {
```

Comment:
when will the replicationTask be nil?

Reply:
for _, event := range history.Events {
			if task.Version != event.GetVersion() {
				return nil, nil
			}
		}